core/src/builder/transaction/challenge.rs

```diff
@@ -127,48 +90,126 @@ pub fn create_watchtower_challenge_txhandler(
         .finalize())
 }
 
-/// Creates a [`TxHandler`] for the `operator_challenge_NACK_tx`. This transaction will force
-/// the operator to reveal the preimage for the corresponding watchtower since if they do not
-/// reveal the preimage, the NofN will be able to spend the output after 0.5 week, which will
-/// prevent the operator from sending `assert_begin_tx`.
-pub fn create_operator_challenge_nack_txhandler(
-    watchtower_challenge_txhandler: &TxHandler,
+pub fn create_watchtower_challenge_txhandler_from_script(
```

Comment:
remove this function entirely and give the script when creating the challenge kickoff